lib/minasu/db.py

```diff
@@ -11,29 +13,67 @@ def __init__(
         self.instance_name = instance_name
         self.instance_dir  = instance_dir
         self.instance_path = self.instance_dir + "/" + self.instance_name
+        self.data = {}
+
+    def lock():
+        '''Temporary method to store lock logic
+
+        '''
+        program = (self.instance_name in (p.name() for p in psutil.process_iter()))
+        lock_dir = '/tmp/minasu/' + self.instance_name
+        if os.path.exists(lock_dir) or program:
+            print("Minasu instance: %s, is already running" % self.instance_name)
+            print("Exiting")
+            sys.exit()
+        else:
+            os.makedirs(lock_dir)
+
 
     # create a new db at folder location
-    def create(self):
-        data = {}
-        # data["instance_name"] = self.instance_name
-        # data["instance_dir"]  = self.instance_dir + "/"
-        # data["instance_path"] = self.instance_path + "/"
+    # def load(self) -> results[JSON]:
+    # create path if doesn't exist
+    # for each dir (bucket)
+    #     for each yml file
+    #         yaml.load()
+
+    def load(self):
+        self.data["instance_name"] = self.instance_name
+        self.data["instance_dir"]  = self.instance_dir + "/"
+        self.data["instance_path"] = self.instance_path + "/"
+        instance = {}
 
         if not os.path.exists(self.instance_path):
             try:
                 os.mkdir(self.instance_path)
             except OSError as e:
                 if e.errno != errno.EEXIST:
                     raise
-            data["created"] = True
+            instance["created"] = True
 
         else:
-            data["created"] = False
+            instance["created"] = False
+            buckets = {}
+            list_of_buckets = os.listdir(self.instance_path)
+            for bucket in list_of_buckets:
+                bucket_path = self.instance_path + "/" + bucket + "/"
+                for item in os.listdir(bucket_path):
+                    if item.lower().endswith(('.yml', '.yaml')):
+                        buckets[bucket] = {}
+                        with open(bucket_path + item, 'r') as stream:
+                            try:
+                                buckets[bucket][item] = yaml.safe_load(stream)
+                            except yaml.YAMLError as exc:
+                                print(exc)
```

Comment:
this needs to also exit with an non zero return code and a friendly message that FILE doesn't contain valid yaml.

Reply:
Exit the program or the try?

Reply:
in the except loop.  by simply printing something, the program will go on.  we (probably) don't want that?

Reply:
I wouldn't want my program to stop because of invalid yml. I say we ignore the invalid yml and write the exception to log, eventually, that these files are corrupted or something to that matter.

Reply:
We want exactly the opposite.  The db itself should be the only thing ever writing to disk.  So if a file load error happens, it's either because of: 
- problem with the code
- problem writing to disk completely (someone pulled the plug during a write?) (this should never happen if the queue class properly writes things to disk in an atomic way)
- someone manually edited a file in an invalid manner

Any of these means that the db is corrupt and needs manual hand holding.  

So yes, we must exit hard and not continue.